src/Concordium/Client/GRPC2.hs

```diff
@@ -0,0 +1,2693 @@
+            -- \|Distribution of transaction fees.
+            _rpTransactionFeeDistribution <- fromProto =<< cParams ^? ProtoFields.transactionFeeDistribution
+            -- \|Rewards paid from the GAS account.
+            _rpGASRewards <- fromProto =<< cParams ^? ProtoFields.gasRewards
+            return Parameters.RewardParameters{..}
+        -- FIXME: ProtoFields.foundationAccount has type AccountAddress, but the type expects an AccountIndex.
+        _cpFoundationAccount <- Nothing
+        _cpPoolParameters <- Nothing
+        return Parameters.ChainParameters{..}
+
+instance FromProto Proto.Epoch where
+    type Output' Proto.Epoch = Epoch
+    fromProto = return . deMkWord64
+
+instance FromProto Proto.CredentialsPerBlockLimit where
+    type Output' Proto.CredentialsPerBlockLimit = CredentialsPerBlockLimit
+    fromProto = deMkWord16
+
+-- FIXME: Pertains to the next FIXME comment.
+data ChainParameterOutput
+    = ChainParameterOutputV0 !(Parameters.ChainParameters' 'ChainParametersV0)
+    | ChainParameterOutputV1 !(Parameters.ChainParameters' 'ChainParametersV1)
+
+instance FromProto Proto.ChainParameters where
+    type
+        -- FIXME: Perhaps there is some nice(r) way achieve this
+        --        with (constrained) type variables or type
+        --        families? But I am uncertain whether the class
+        --        has to change, or if this can be done "locally"?
+        --        Or does it not matter?
+        Output' Proto.ChainParameters =
+            ChainParameterOutput
+    fromProto cParams = do
+        cp <- cParams ^. Proto.maybe'parameters
+        case cp of
+            Proto.ChainParameters'V0 v0 -> ChainParameterOutputV0 <$> fromProto v0
+            Proto.ChainParameters'V1 v1 -> ChainParameterOutputV1 <$> fromProto v1
+
+data VersionedModuleSource
+    = ModuleSourceV0 !ByteString
+    | ModuleSourceV1 !ByteString
+
+type AccountTransactionSignHash = ByteString
+
+instance FromProto Proto.AccountTransactionSignHash where
+    type Output' Proto.AccountTransactionSignHash = AccountTransactionSignHash
+    fromProto atsHash = atsHash ^? ProtoFields.value
+
+data AccountTransactionPayload
+    = Raw !ByteString
+    | VersionedModuleSourcePayload
+        !VersionedModuleSource
+    | InitContractPayload
+        { amount :: !Amount
+        , moduleRef :: !ModuleRef
+        , initName :: !Wasm.InitName
+        , parameter :: !Wasm.Parameter
+        }
+    | UpdateContractPayload
+        { amount :: !Amount
+        , address :: !ContractAddress
+        , receiveName :: !Wasm.ReceiveName
+        , parameter :: !Wasm.Parameter
+        }
+    | TransferPayload
+        { amount :: !Amount
+        , receiver :: !AccountAddress
+        }
+    | TransferWithMemoPayload
+        { amount :: !Amount
+        , receiver :: !AccountAddress
+        , memo :: !Memo
+        }
+    | RegisteredDataPayload
+        !RegisteredData
```

Comment:
We don't need this. This is for cases when there is no SDK. We have the full transaction payload in Haskell.

Reply:
Some more context:
When you want to send a transaction, you need to sign the payload and include the signature. The issue is that it is the _serialized_ and hashed payload that you need to sign and the serialization must match the one used in the node. So if you create a transaction with the protobuf datatypes and serialize that, you will get a different payload (because protobuf has a different way of serializing the data). So you will end up signing the wrong payload, making the transaction invalid.
That is why we have the endpoint with `PreAccountTransaction` for getting the correct hash to sign for the transaction. 
But if you have data types for the transaction, which are serialized in the way the node expects, then you can get the right payload directly and sign it. That is what you should do here with the client.